src/plugins/ui_core_views/evaluation.ts

```diff
@@ -60,48 +63,118 @@ export class EvaluationPlugin extends UIPlugin {
     "getEvaluatedCellsInZone",
   ] as const;
 
+  private isFormulaDependenciesUpToDate = false;
   private isUpToDate = false;
-  private evaluatedCells: {
-    [sheetId: UID]:
-      | {
-          [col: HeaderIndex]: { [row: HeaderIndex]: Lazy<EvaluatedCell> | undefined } | undefined;
-        }
-      | undefined;
-  } = {};
   private readonly evalContext: EvalContext;
-  private readonly lazyEvaluation: boolean;
+
+  private evaluatedCells: PositionDict<EvaluatedCell> = {};
+  private nextXcsToUpdate: Set<string> = new Set<string>();
+  private currentXcsToUpdate: Set<string> = new Set<string>();
+
+  private formulaDependencies = new Graph();
+  private spreadDependencies = new Graph();
+  private spreadCandidates = new BiDirectionnalGraph();
```

Comment:
It's really important to document these variables. 
I think we should write a documentation on how it works, what are the choices we made, ...
Something like we did for collaborative edition